### PR TITLE
Add rust wrappers for the sgx attributes types

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Add rust wrappesr for:
- sgx_attributes_t => Attributes
- sgx_misc_select_t => MiscellaneousSelect
- sgx_misc_attribute_t => MiscellaneousAttribute

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #82** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/82" target="_blank">Graphite</a> 
    * **PR #85** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/85" target="_blank">Graphite</a> 
      * **PR #86** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86?utm_source=stack-comment).

File: core/sys/types/src/lib.rs [hunk 1]
```diff
@@ -10,4 +10,21 @@
     non_upper_case_globals
 )]
 
+use core::fmt;
 include!(concat!(env!("OUT_DIR"), "/bindings.rs"));
+
+impl fmt::Debug for sgx_attributes_t {
```
Comment (nick-mobilecoin) [inline core/sys/types/src/lib.rs:16]: I'm not fond of this, I wanted to have the tests to show the wrapping worked.
I want to look into forcing bindgen to generate debug and Eq for structs, not sure why it's not, it seems to only be doing it for simple primitive aliases

**Review by awygle — APPROVED** [review]
This looks fine to me. Arguably the Debug and PartialEq implementations should be tested, but I'm not religious about that, especially if we're going to eventually offload that work to bindgen.

**Comment by nick-mobilecoin** [comment]
> This looks fine to me. Arguably the Debug and PartialEq implementations should be tested, but I'm not religious about that, especially if we're going to eventually offload that work to bindgen.

I just want to say 100% on testing the functionality, but hopefully I can get the derive working through bindgen so it would be wasted right now.

**Review by jcape — APPROVED** [review]

File: core/sys/types/src/lib.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply core/sys/types/src/lib.rs]: this is taken care of in #91

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86?utm_source=stack-comment).


Job Reference: 2v9KzWSc3RZmb6aSw9co

**Pull request merged**